model_compression_toolkit/core/common/similarity_analyzer.py

```diff
@@ -41,74 +41,75 @@ def tensor_norm(x: np.ndarray, p: float = 2.0) -> np.float:
+    Flattening the samples batch to allow similarity analysis computation per sample.
 
     Args:
-        float_tensor: First tensor to compare.
-        fxp_tensor: Second tensor to compare.
-        norm: whether to normalize the error function result.
-        norm_eps: epsilon value for error normalization stability.
-
-    Returns:
-        The MSE distance between the two tensors.
-    """
-    validate_before_compute_similarity(float_tensor, fxp_tensor)
-    error = np.power(float_tensor - fxp_tensor, 2.0).mean()
-    if norm:
-        error /= (np.power(float_tensor, 2.0).mean() + norm_eps)
-    return error
+        t: A tensor to be flattened.
+        batch: Whether the similarity computation is per image or per tensor.
 
+    Returns: A flattened tensor which has the number of samples as is first dimension.
 
-def compute_nmse(float_tensor: np.ndarray, fxp_tensor: np.ndarray) -> float:
     """
-    Compute the normalized mean square error between two numpy arrays.
 
-    Args:
-        float_tensor: First tensor to compare.
-        fxp_tensor: Second tensor to compare.
+    if batch:
+        f_t = t.reshape([t.shape[0], -1])
+    else:
+        f_t = t.reshape([1, -1])
 
-    Returns:
-        The NMSE distance between the two tensors.
-    """
-    validate_before_compute_similarity(float_tensor, fxp_tensor)
-    normalized_float_tensor = float_tensor / tensor_norm(float_tensor)
-    normalized_fxp_tensor = fxp_tensor / tensor_norm(fxp_tensor)
-    return np.mean(np.power(normalized_float_tensor - normalized_fxp_tensor, 2.0))
+    return f_t
+
+#########################
+# Similarity functions
+#########################
 
 
-def compute_nmae(float_tensor: np.ndarray, fxp_tensor: np.ndarray) -> float:
+def compute_mse(float_tensor: np.ndarray,
+                fxp_tensor: np.ndarray,
+                norm: bool = False,
+                norm_eps: float = 1e-8,
+                batch: bool = False) -> float:
     """
-    Compute the normalized mean average error between two numpy arrays.
+    Compute the mean square error between two numpy arrays.
 
     Args:
         float_tensor: First tensor to compare.
         fxp_tensor: Second tensor to compare.
+        norm: whether to normalize the error function result.
+        norm_eps: epsilon value for error normalization stability.
+        batch: Whether to run batch similarity analysis or not.
 
     Returns:
-        The NMAE distance between the two tensors.
+        The MSE distance between the two tensors.
     """
     validate_before_compute_similarity(float_tensor, fxp_tensor)
-    normalized_float_tensor = float_tensor / tensor_norm(float_tensor, 1.0)
-    normalized_fxp_tensor = fxp_tensor / tensor_norm(fxp_tensor, 1.0)
-    return np.mean(np.abs(normalized_float_tensor - normalized_fxp_tensor))
+
+    float_flat = flatten_tensor(float_tensor, batch)
+    fxp_flat = flatten_tensor(fxp_tensor, batch)
+
+    error = ((float_flat - fxp_flat) ** 2).mean(axis=-1)
```

Comment:
Maybe here you can reuse code and use tensor_norm.

Reply:
I'm not sure that it performs a similar operation - 
(the computation on the i'th row of tensor x)

here: SUM(([x^2]_i)) / |x_i|
tensor_norm: SQRT(SUM([x^2]_i))

Am I missing something?

Reply:
MSE = norm square. The normalize 1/|x_i| is configurable in mse computation.

Reply:
Not sure I understand, what call are you suggesting that I put here? (also, is it necessary? I don't think that the potential code duplication here is problematic, and it might be better to keep the code simple and declarative as it is right now)

Reply:
Ok. let's leave it.